pyiceberg/serializers.py

```diff
@@ -127,6 +127,6 @@ def table_metadata(metadata: TableMetadata, output_file: OutputFile, overwrite:
             overwrite (bool): Where to overwrite the file if it already exists. Defaults to `False`.
         """
         with output_file.create(overwrite=overwrite) as output_stream:
-            json_bytes = metadata.model_dump_json().encode(UTF8)
+            json_bytes = metadata.model_dump_json(exclude_none=False).encode(UTF8)
```

Comment:
Maybe not a big deal: Shall we set `exclude_none=False` only in the legacy mode?

Reply:
I would prefer that. Let me put in this update